src/sbvr-api/permissions.ts

```diff
@@ -607,24 +607,20 @@ const generateConstrainedAbstractSql = (
 			parentOdata,
 		);
 
-		if (collapsedParentPermissionFilters === false) {
-			// We reuse a constant permission error here as it will be cached, and
-			// using a single error instance can drastically reduce the memory used
-			throw constrainedPermissionError;
-		}
-
-		rewriteSubPermissionBindings(
-			collapsedParentPermissionFilters,
-			this.bindVarsLength + this.extraBindVars.length,
-		);
-		convertToLambda(collapsedParentPermissionFilters, lambdaId);
-		resolvedProperty.lambda = {
-			method: 'any',
-			identifier: lambdaId,
-			expression: collapsedParentPermissionFilters,
-		};
+		if (collapsedParentPermissionFilters != null) {
```

Comment:
I think you can immediately return `['Equals', ['Boolean', true], ['Boolean', true]]` to avoid the extra work and subquery